content/en/blog/_posts/2023-06-29-container-image-signature-verification/index.md

```diff
@@ -0,0 +1,305 @@
+validated:
+
+```console
+DEBU[…] IsRunningImageAllowed for image docker:quay.io/crio/signed:latest
+DEBU[…]  Using transport "docker" specific policy section quay.io/crio/signed
+DEBU[…] Reading /var/lib/containers/sigstore/crio/signed@sha256=18b42e8ea347780f35d979a829affa178593a8e31d90644466396e1187a07f3a/signature-1
+DEBU[…] Looking for sigstore attachments in quay.io/crio/signed:sha256-18b42e8ea347780f35d979a829affa178593a8e31d90644466396e1187a07f3a.sig
+DEBU[…] GET https://quay.io/v2/crio/signed/manifests/sha256-18b42e8ea347780f35d979a829affa178593a8e31d90644466396e1187a07f3a.sig
+DEBU[…] Content-Type from manifest GET is "application/vnd.oci.image.manifest.v1+json"
+DEBU[…] Found a sigstore attachment manifest with 1 layers
+DEBU[…] Fetching sigstore attachment 1/1: sha256:8276724a208087e73ae5d9d6e8f872f67808c08b0acdfdc73019278807197c45
+DEBU[…] Downloading /v2/crio/signed/blobs/sha256:8276724a208087e73ae5d9d6e8f872f67808c08b0acdfdc73019278807197c45
+DEBU[…] GET https://quay.io/v2/crio/signed/blobs/sha256:8276724a208087e73ae5d9d6e8f872f67808c08b0acdfdc73019278807197c45
+DEBU[…]  Requirement 0: allowed
+DEBU[…] Overall: allowed
+```
+
+All of the defined fields like `oidcIssuer` and `subjectEmail` in the policy
+have to match, while `fulcio.caData` and `rekorPublicKeyData` are the public
+keys from the upstream [fulcio (OIDC PKI)][fulcio] and [rekor
+(transparency log)][rekor] instances.
+
+[fulcio]: https://github.com/sigstore/fulcio
+[rekor]: https://github.com/sigstore/rekor
+
+This means that if you now invalidate the `subjectEmail` of the policy, for example to
+`wrong@mail.com`:
+
+```console
+> jq '.transports.docker."quay.io/crio/signed"[0].fulcio.subjectEmail = "wrong@mail.com"' policy.json > new-policy.json
+> mv new-policy.json policy.json
+```
+
+Then remove the image, since it already exists locally:
+
+```console
+> sudo crictl rmi quay.io/crio/signed
+```
+
+Now when you pull the image, CRI-O complains that the required email is wrong:
+
+```console
+> sudo crictl pull quay.io/crio/signed
+FATA[…] pulling image: rpc error: code = Unknown desc = Source image rejected: Required email wrong@mail.com not found (got []string{"sgrunert@redhat.com"})
+```
+
+It is also possible to test an unsigned image against the policy. For that you
+have to modify the key `quay.io/crio/signed` to something like
+`quay.io/crio/unsigned`:
+
+```console
+> sed -i 's;quay.io/crio/signed;quay.io/crio/unsigned;' policy.json
+```
+
+If you now pull the container image, CRI-O will complain that no signature exists
+for it:
+
+```console
+> sudo crictl pull quay.io/crio/unsigned
+FATA[…] pulling image: rpc error: code = Unknown desc = SignatureValidationFailed: Source image rejected: A signature was required, but no signature exists
+```
+
+It's important to mention that CRI-O will match the
+`.critical.identity.docker-reference` field within the signature to match with
+the image repository. For example, if I verify the image
```

Comment:
```suggestion
the image repository. For example, if you verify the image
```

Reply:
Thank you @saschagrunert for incorporating the suggestions. Just a couple more for the latest paragraph that you added and we should be good to go. 

Overall, this LGTM. @sftim or @reylejano or @nate-double-u, would you care to add your comment(s) and/or the LGTM label to this PR? Thanks!